### ISSUE TITLE
Add solr support for synonyms for numbers/abbreviations

### ISSUE THREAD

**Issue by popcar2** [internetarchive/openlibrary#6635]
I've been using the website for a long time now and one of my biggest gripes is how searching works. When searching for books in OpenLibrary, you often need to write exactly the correct title. This means that if a book uses words for numbers (One, Two, Three etc), searching the same title with digits (1, 2, 3 etc) would give no result.

Another example is if a book uses "Vol." in the title, searching "volume" would net no result even though they mean the same thing. This makes finding specific books a lot more difficult.

### Describe the problem that you'd like solved
The search engine searches exact terms, but it should have tolerance when dealing with numbers or words of equivalent meaning. 
Here's an example:

![image](https://user-images.githubusercontent.com/16920817/172428012-a37c71ef-9d8e-4e20-af48-d76affc530a0.png)
![image](https://user-images.githubusercontent.com/16920817/172428086-29953c73-c6c5-44a9-bfdf-5984fcd14c37.png)
I would like searching "The Walking Dead Compendium Four" and "The Walking Dead Compendium 4" to find the book.

### Proposal & Constraints
The search engine should be error tolerant to words of the same meaning. 
"Vol." should be the same as writing "Volume"
"Two" should be the same as writing "2" or "II"
"&" and "and" should also be interchangeable.

### Additional context
Another example, but with "vol" and "volume"
![image](https://user-images.githubusercontent.com/16920817/172433397-c8cc2773-5991-454f-a1af-feb52c40d311.png)
![image](https://user-images.githubusercontent.com/16920817/172433433-9cbeebc8-2b6d-45cd-bdff-0447f661978d.png)

**Comment by cdrini** [comment]
I think the solution for this would be to make use of solr's synonyms feature. But some experimenting / investigation needed. Anyone who has some time to experiment with adding synonyms to solr, please do!

**Comment by bicolino34** [comment]
@cdrini I would like to do it, how can I?

**Comment by bicolino34** [comment]
The search is strict not only with terms, but also with letters. Compare [Безпека життєдіяльності](https://openlibrary.org/search?q=%D0%91%D0%B5%D0%B7%D0%BF%D0%B5%D0%BA%D0%B0+%D0%B6%D0%B8%D1%82%D1%82%D1%94%D0%B4%D1%96%D1%8F%D0%BB%D1%8C%D0%BD%D0%BE%D1%81%D1%82%D1%96&mode=everything) and [Безпека життєдіяльност](https://openlibrary.org/search?q=%D0%91%D0%B5%D0%B7%D0%BF%D0%B5%D0%BA%D0%B0+%D0%B6%D0%B8%D1%82%D1%82%D1%94%D0%B4%D1%96%D1%8F%D0%BB%D1%8C%D0%BD%D0%BE%D1%81%D1%82&mode=everything). With just one letter missing (і) there are no results

**Comment by cdrini** [comment]
So this is a solr research task; here are some of places where it will need modifications:

The solr schema which defines the various type of text fields has synonyms enabled -- but only at query time:

https://github.com/internetarchive/openlibrary/blob/82bc2f61c8c41363567d398b7b027a16775dbc91/conf/solr/conf/managed-schema#L426-L467

This blog post has some info: https://library.brown.edu/create/digitaltechnologies/using-synonyms-in-solr/

In a nutshell we need synonyms inside https://github.com/internetarchive/openlibrary/blob/ccabd95be2a82c4f79d94b1f10e46ea1d3c5c730/conf/solr/conf/synonyms.txt

And then test locally with a full reindex (See https://github.com/internetarchive/openlibrary/wiki/Solr#making-changes-to-solr-config )

But for numbers, they probably need to be in English only for now? I'm not sure how we should handle non-English numbers. Ideally we'd want different synonyms files for different user locales, but I'm not sure if/how to do this in solr.

**Comment by cdrini** [comment]
But we can definitely add something like `vol,vol.,Volume` in there and see if it helps with that!

**Comment by cdrini** [comment]
Actually it looks like the synonyms file is working! You can see the `TV` one in action here: https://openlibrary.org/search?q=television+kid&mode=everything . 

So adding volume should be easy enough!

**Comment by mekarpeles** [comment]
Is this addressed by #6922? Can this issue be consolidated into that one?

**Comment by tfmorris** [comment]
#6922 is the PR which is meant to address this issue, but it's stuck in review and has some issues.

**Comment by cdrini** [comment]
Yep; if someone with solr knowledge has the time to take that branch and clean it up and test it, that would be a huge help!
